distributed/worker.py

```diff
@@ -1649,10 +1647,11 @@ def _close(executor, wait):
         await self.rpc.close()
 
         self.status = Status.closed
+        setproctitle("dask worker [closed]")
+
         await ServerNode.close(self)
 
         self.__exit_stack.__exit__(None, None, None)
```

Comment:
Just a side comment: The cleanest way would probably be to have everything be contextaware such that we can throw everything into the exitstack